dot/node.go

```diff
@@ -171,28 +215,32 @@ func LoadGlobalNodeName(basepath string) (nodename string, err error) {
 	nodename, err = basestate.LoadNodeGlobalName()
 	if err != nil {
 		logger.Warnf("failed to load global node name from base path %s: %s", basepath, err)
-		return "", err
 	}
-
 	return nodename, err
 }
 
-// NewNode creates a new dot node from a dot node configuration
+// NewNode to create node based on given Config
 func NewNode(cfg *Config, ks *keystore.GlobalKeystore) (*Node, error) {
+	return newNode(cfg, ks, nodeBuilder{})
```

Comment:
Perhaps the builder could be injected as an (exported) interface to `NewNode` and `InitialiseNode`, that would be nice.

Reply:
I don't understand what you mean here.